### PR TITLE
Login register feature and full redirection functionality

### PR DESCRIPTION
**Pull request by PatrykDu**

### CODE REVIEW

File: routers/auth.py [hunk 1]
```diff
@@ -17,19 +24,135 @@
 )
 
 
-def get_db():
-    try:
-        db = SessionLocal()
-        yield db
-    finally:
-        db.close()
+def authenticate_user(username: str, password: str, db):
```
Comment (Marvo5) [inline routers/auth.py:27]: missing docstring

File: routers/auth.py [hunk 2]
```diff
@@ -17,19 +24,135 @@
...
+        models.User.username == username).first()
+
+    if not user:
+        return False
+    if password != user.hashed_password:
+        return False
+    return user
+
+
+def create_access_token(username: str, user_id: int,
```
Comment (Marvo5) [inline routers/auth.py:38]: missing docstring

File: templates/admin.html [hunk 3]
```diff
@@ -0,0 +1 @@
+{% include 'layout.html' %}
```
Comment (Marvo5) [inline templates/admin.html:1]: image

vs

image

File: templates/customer.html [hunk 4]
```diff
@@ -0,0 +1 @@
+{% include 'layout.html' %}
```
Comment (Marvo5) [inline templates/customer.html:1]: [image](https://private-user-images.githubusercontent.com/113666877/312248259-f3eb32f0-2b21-4f91-be99-1c067b5b2376.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyODA5NTUsIm5iZiI6MTcxMDI4MDY1NSwicGF0aCI6Ii8xMTM2NjY4NzcvMzEyMjQ4MjU5LWYzZWIzMmYwLTJiMjEtNGY5MS1iZTk5LTFjMDY3YjViMjM3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQyMTU3MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDQ5NzAxYjhjNGNhOTRhNmRmMzI2MGJlMTZlYWM0YmY4NDE2YTAyZjIzMGJiMGJkZDMzMmI1YjU1NjZlMjY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.26JZpvs3QKf6y4vOafzDKO50VN7I_Se-YLUEss9iSp8)

vs

[image](https://private-user-images.githubusercontent.com/113666877/312248367-587164b4-dff6-4dcd-a078-7cfaf6ed9719.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyODA5NTUsIm5iZiI6MTcxMDI4MDY1NSwicGF0aCI6Ii8xMTM2NjY4NzcvMzEyMjQ4MzY3LTU4NzE2NGI0LWRmZjYtNGRjZC1hMDc4LTdjZmFmNmVkOTcxOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQyMTU3MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZWMzMjliNTJhNTgyMWQ3YTI1Yjc1YjMwYWU3MGI0YjQ5ZmE2ODJjYWNiM2UzMDk2ZDZjOWE5YzMzMjE2MDg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fvLqcMCxOAyV0UPega-i34goMmm9J0kOBuoVr3IDXfg)

File: templates/mechanic.html [hunk 5]
```diff
@@ -0,0 +1 @@
+{% include 'layout.html' %}
```
Comment (Marvo5) [inline templates/mechanic.html:1]: [image](https://private-user-images.githubusercontent.com/113666877/312248259-f3eb32f0-2b21-4f91-be99-1c067b5b2376.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyODA5NTUsIm5iZiI6MTcxMDI4MDY1NSwicGF0aCI6Ii8xMTM2NjY4NzcvMzEyMjQ4MjU5LWYzZWIzMmYwLTJiMjEtNGY5MS1iZTk5LTFjMDY3YjViMjM3Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQyMTU3MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNDQ5NzAxYjhjNGNhOTRhNmRmMzI2MGJlMTZlYWM0YmY4NDE2YTAyZjIzMGJiMGJkZDMzMmI1YjU1NjZlMjY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.26JZpvs3QKf6y4vOafzDKO50VN7I_Se-YLUEss9iSp8)

vs

[image](https://private-user-images.githubusercontent.com/113666877/312248367-587164b4-dff6-4dcd-a078-7cfaf6ed9719.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAyODA5NTUsIm5iZiI6MTcxMDI4MDY1NSwicGF0aCI6Ii8xMTM2NjY4NzcvMzEyMjQ4MzY3LTU4NzE2NGI0LWRmZjYtNGRjZC1hMDc4LTdjZmFmNmVkOTcxOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxMlQyMTU3MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZWMzMjliNTJhNTgyMWQ3YTI1Yjc1YjMwYWU3MGI0YjQ5ZmE2ODJjYWNiM2UzMDk2ZDZjOWE5YzMzMjE2MDg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fvLqcMCxOAyV0UPega-i34goMmm9J0kOBuoVr3IDXfg)

File: routers/auth.py [hunk 6]
```diff
@@ -17,19 +24,135 @@
...
+                                expires_delta=token_expires)
+
+    response.set_cookie(key="access_token", value=token, httponly=True)
+
+    return True
 
 
 @router.get("/login", response_class=HTMLResponse)
 async def login_page(request: Request):
+
```
Comment (Marvo5) [inline routers/auth.py:68]: remove newline

File: routers/auth.py [hunk 7]
```diff
@@ -17,19 +24,135 @@
...
+            return templates.TemplateResponse("login.html", {"request": request, "msg": msg, "user": user_model})
+        return response
+    except HTTPException:
+        msg = "Unknown Error"
+        return templates.TemplateResponse("login.html", {"request": request, "msg": msg})
+
+
 @router.get("/register", response_class=HTMLResponse)
 async def login_page(request: Request):
+
```
Comment (Marvo5) [inline routers/auth.py:109]: remove newline

File: routers/auth.py [hunk 8]
```diff
@@ -17,19 +24,135 @@
...
+        return RedirectResponse(url="/", status_code=status.HTTP_302_FOUND)
     return templates.TemplateResponse("register.html", {"request": request})
+
+
+@router.post("/register", response_class=HTMLResponse)
+async def register_user(request: Request, email: str = Form(...), username: str = Form(...),
+                        firstname: str = Form(...), lastname: str = Form(...),
+                        password: str = Form(...), password2: str = Form(...),
+                        db: Session = Depends(get_db)):
+
```
Comment (Marvo5) [inline routers/auth.py:121]: remove newline

File: routers/customer.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+router = APIRouter(
+    prefix="/customer",
+    tags=["customer"],
+    responses={401: {"user": "Not authorized"}}
+)
+
+
+@router.get("/", response_class=HTMLResponse)
+async def home_page(request: Request, db: Session = Depends(get_db)):
+
```
Comment (Marvo5) [inline routers/customer.py:24]: remove newline

File: routers/mechanic.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+router = APIRouter(
+    prefix="/mechanic",
+    tags=["mechanic"],
+    responses={401: {"user": "Not authorized"}}
+)
+
+
+@router.get("/", response_class=HTMLResponse)
+async def home_page(request: Request, db: Session = Depends(get_db)):
+
```
Comment (Marvo5) [inline routers/mechanic.py:23]: remove newline

**Review by Marvo5 — CHANGES_REQUESTED** [review]
Missing content in navbar

File: templates/admin.html (hunk 3 above)
Comment (PatrykDu) [reply templates/admin.html:1]: It suppose to be like that. We don't want to be able to send message as a logged user.

File: routers/mechanic.py (hunk 10 above)
Comment (PatrykDu) [reply routers/mechanic.py:23]: There always suppose to be new line on the end of the .py file. xD

File: templates/mechanic.html (hunk 5 above)
Comment (PatrykDu) [reply templates/mechanic.html:1]: image doesnt work

File: routers/auth.py (hunk 6 above)
Comment (PatrykDu) [reply routers/auth.py:68]: ?

File: routers/auth.py (hunk 7 above)
Comment (PatrykDu) [reply routers/auth.py:109]: there suppose to be a new line

File: templates/customer.html (hunk 4 above)
Comment (PatrykDu) [reply templates/customer.html:1]: image doesnt work

File: routers/auth.py (hunk 8 above)
Comment (PatrykDu) [reply routers/auth.py:121]: as above

File: routers/customer.py (hunk 9 above)
Comment (PatrykDu) [reply routers/customer.py:24]: as above

File: routers/auth.py (hunk 1 above)
Comment (PatrykDu) [reply routers/auth.py:27]: added

File: routers/auth.py (hunk 2 above)
Comment (PatrykDu) [reply routers/auth.py:39]: added

**Comment by PatrykDu** [comment]
> Missing content in navbar

Contact form button is suppose to be missing after user is logged it. Contact for has no option to connect logged in user to the sent data, so it should be available only for not logged in users as a one of the contact form (the other ones are in the footer). After user is logged in, he can use calendar or talk with mechanic in the repair section

**Review by Marvo5 — APPROVED** [review]
OK

**Pull request merged**